packages/core/echo/echo/src/struggle.test.ts

```diff
@@ -0,0 +1,126 @@
+import { Schema } from 'effect';
+/// ISSUE(dm): Is it okay to have one entry point for all of the types like object id and type definitions and will it prohibit us from building isolated storybooks that don't pull in echo? DX vs tree shaking.
+// So that type and relation API should be separate from echo database API.
+// Potentially, some part of the database API could be included in there. Things like query builders. So only the declarative API, not the actual execution layer.
+import { test } from 'vitest';
+import { Type } from '.';
+
+declare const create: any;
+declare const Relation: any;
+declare const setupDb: any;
+declare const Filter: any;
+declare const Query: any;
+
+///
+
+const Contact = Schema.Struct({
+  // ISSUE(dm): Forcing explicit object IDs for every schema?
+  id: Type.ObjectId,
+  name: Schema.String,
+  // ISSUE(dm): Reconcile "type" and "typename" i.e. the dxn and the name of the type.
+}).pipe(Type.def({ typename: 'dxos.org/type/Contact', version: '0.1.0' }));
+
+const Org = Schema.Struct({
+  id: Type.ObjectId,
+  name: Schema.String,
+}).pipe(Type.def({ typename: 'dxos.org/type/Org', version: '0.1.0' }));
+
+const WorksFor = Schema.Struct({
+  id: Type.ObjectId,
+  since: Schema.String,
+  ...Relation.connects(Contact, Org),
+  // ISSUE(dm): "dxos.org/type" vs "dxos.org/relation"
+}).pipe(Relation.def({ typename: 'dxos.org/type/WorksFor', version: '0.1.0' }));
+
+test('db', async () => {
+  const db = setupDb();
+
+  const contact = db.add(
+    create(Contact, {
+      name: 'John Doe',
+    }),
+  );
+
+  const org = db.add(
+    create(Org, {
+      name: 'Acme Inc.',
+    }),
+  );
+
+  const worksFor = db.add(
+    create(WorksFor, {
+      since: '2021-01-01',
+      ...Relation.make(contact, org),
+      ...Type.meta({
+        keys: [],
+      }),
+    }),
+  );
+
+  {
+    // ISSUE(dm): Don't like get API. This won't return an object if it's not loaded.
```

Comment:
let's evolve it. perhaps a wrapper. but suggest getting the rest working first. i.e., second phase.